### PR TITLE
Extend gramex.cache.open to read named ranges and workbook tables #309

### PR DESCRIPTION
**Pull request by sanand0**

### CODE REVIEW

File: gramex/cache.py [hunk 1]
```diff
@@ -128,6 +128,45 @@ def _template(path, **kwargs):
...
+        range_str = wb.defined_names.get(name, sheetid).attr_text.split('!')[1].replace('$', '')
+        get_rows(sheet_name, range_str)
+
+    if table is not None:
+        get_rows(sheet_name, wb[sheet_name].tables[table].ref)
+
+    header_val = None
+    if header is not None:
+        header_val = list(data_rows[header])
+        del data_rows[header]
```
Comment (pratapvardhan) [inline gramex/cache.py:164]: @vinothsm -- I suspect if this handles `header can be list of int` scenario. If you're handling only int and None, you could potential simplify with .. ?

```python
header = None if header is None else list(data_rows.pop(header))
df = pd.DataFrame(data_rows, columns=header)
```
without needing to `del` and intermediate `header_val`.

File: gramex/cache.py [hunk 2]
```diff
@@ -128,6 +128,45 @@ def _template(path, **kwargs):
...
 
 
+def excel_reader(path, sheet_name='Sheet1', range=None, name=None, table=None, header=0):
+    # import openpyxl
+    # Read the file using openpyxl
+    # If range (e.g. A10:C30) is specified, return cells in that range
+
+    # If name is specified, return cells in that named range using wb.defined_names
+    # If table is specified, return cells in that named table using ws.tables
+    # Convert into a DataFrame -- using header if specified. header can be int, list of int, None
```
Comment (pratapvardhan) [inline gramex/cache.py:138]: What would happen if you use both `range` and `name` and or `table`? If it's only intended to handle exclusive cases, `if, else if` control logic might be appropriate here no?

File: gramex/cache.py (hunk 1 above)
Comment (vinothsm) [reply gramex/cache.py:164]: @pratapvardhan  Thanks for reviewing. `header` is int only and agree with your suggestion to simplify like above.

File: gramex/cache.py (hunk 2 above)
Comment (vinothsm) [reply gramex/cache.py:138]: @pratapvardhan We are yet to decide the behavior in such scenarios. @sanand0 Could you please provide your thoughts?

Comment (sanand0) [reply gramex/cache.py:138]: table overrides name overrides range. I've implemented this

File: gramex/cache.py (hunk 1 above)
Comment (sanand0) [reply gramex/cache.py]: Have implemented list of int as a possibility.

**Pull request merged**